src/iam.ts

```diff
@@ -580,7 +580,7 @@ export class IAM extends IAMBase {
             },
         };
         const jwt = new JWT(new Wallet(delegateKey));
-        const identityToken = jwt.sign(payload, { algorithm: "ES256" });
+        const identityToken = jwt.sign(payload, { algorithm: "ES256", issuer: identity });
```

Comment:
side question: how is the algorithm type deterministic to the token creation / verification ?
Would a use of "HS256" instead of "ES256" lead to problems ?